tools/install_bazel.sh

```diff
@@ -6,9 +6,17 @@ echo "Checking if Bazel ${BAZEL_VERSION} needs to be installed..."
 if [[ $(command -v bazel) && "$(bazel version | grep 'label' | cut -d ' ' -f 3)" =~ ${BAZEL_VERSION} ]]; then
   echo "Bazel ${BAZEL_VERSION} already installed, skipping fetch."
 else
-  wget -nv https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-installer-linux-x86_64.sh
-  chmod +x bazel-${BAZEL_VERSION}-installer-linux-x86_64.sh
-  sudo -E ./bazel-${BAZEL_VERSION}-installer-linux-x86_64.sh
-  sudo -E mv /usr/local/bin/bazel /usr/bin
-  rm bazel-${BAZEL_VERSION}-installer-linux-x86_64.sh
+  if [ "$(uname -m)" = "aarch64" ]; then
+     echo "Install Bazel on aarch64 ..."
+     wget -nv https://github.com/Tick-Tocker/bazel-arm64/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-linux-aarch64
```

Comment:
This Bazel source appears to be a random github user. Are there official builds we can pull from somewhere? Is there any ongoing effort to upstream any arm64-specific changes?

Reply:
Bazel upstream doesn't have any official aarch64 builds, but we have aarch64 builds in openSUSE:

https://build.opensuse.org/package/binaries/devel:tools:building/bazel2.0/openSUSE_Factory_ARM

OBS also allows to create deb packages, or rpm which works on CentOS/RHEL/Fedora. Maybe I can create a project/repo with aarch64 builds for different distros.

Or maybe just unpacking the openSUSE rpm will work? Bazel is written in Java, so it should not depend on any shared libraries.

Reply:
@joestringer This bazel-arm64 [link](https://github.com/Tick-Tocker/bazel-arm64/releases) is not a so random github user, it maintains a bazel arm64 binary release series from 1.1.0 to 2.2.0 which are up-to-date as the official releases except the most latest version(3.0.0) and currently Bazel only needs the binary to run which does not depend on the shared lib.

On the other side, there are several options in case it's not available in the future even for the availability of Bazel arm64:
1. Build the Bazel from source, which is not so complex and actually I have the script to build it. Now I just think it's not so urgent to build it from source which may cost a little bit more time;
2. Upstream the Bazel arm64 binary to some stable repositories, for example, we can fork the Bazel into the Cilium project and upstream the binary releases there. Here we only need the binary of Bazel, not a rpm or deb pkg, which installation relies on a certain Linux release, e.g., CentOS/Redhat/Ubuntu/Debian.
3. The official support of arm64 from Bazel [offical repo](https://github.com/bazelbuild/bazel/) would be available in the future. We can change it to that repo when it's ok.

So I think it's ok to use this solution here currently and we are still confident of supporting it well in the future when things changed.

Reply:
I had added an issue in the bazel official repo for Bazel arm64 support:
https://github.com/bazelbuild/bazel/issues/11134

Reply:
Bazel providing arm64 or non-x86 architecture builds has been a long standing problem, which I don't see being resolved any time soon.  This same issue was opened up 10 months ago: https://github.com/bazelbuild/bazel/issues/8833

Reply:
echo "deb https://download.opensuse.org/repositories/home:/mrostecki:/bazel/xUbuntu_20.04/ /" > /etc/apt/sources.list.d/bazel.list"
curl -L https://download.opensuse.org/repositories/home:/mrostecki:/bazel/xUbuntu_20.04/Release.key | sudo apt-key add -
apt update
apt install bazel

Reply:
this needs to be addressed @TrevorTaoARM. Everything should be ready Michal in OpenSuse.